homeassistant/components/rki_covid/sensor.py

```diff
@@ -0,0 +1,212 @@
+            district_sensors = [
+                RKICovidNumbersSensor(coordinator, district, info_type)
+                for info_type in DISTRICT_SENSORS
+            ]
+            async_add_entities(district_sensors, update_before_add=True)
+
+
+async def async_setup_entry(
+    hass: core.HomeAssistant,
+    config_entry: config_entries.ConfigEntry,
+    async_add_entities,
+):
+    """Create sensors from a config entry in the integrations UI."""
+    _LOGGER.debug(f"create sensor from config entry {config_entry.data}")
+    session = async_get_clientsession(hass)
+    parser = RkiCovidParser(session)
+    coordinator = await get_coordinator(hass, parser)
+
+    if coordinator is None or coordinator.data is None:
+        raise PlatformNotReady("Data coordinator could not be initialized!")
+
+    district = config_entry.data[ATTR_COUNTY]
+    sensors = [
+        RKICovidNumbersSensor(coordinator, district, info_type) for info_type in SENSORS
+    ]
+    async_add_entities(sensors, update_before_add=True)
+
+    # add additional sensors for districts
+    if district.startswith("BL"):
+        district_sensors = [
+            RKICovidNumbersSensor(coordinator, district, info_type)
+            for info_type in DISTRICT_SENSORS
+        ]
+        async_add_entities(district_sensors, update_before_add=True)
+
+
+class RKICovidNumbersSensor(CoordinatorEntity):
+    """Representation of a sensor."""
+
+    name = None
+    unique_id = None
+
+    def __init__(
+        self,
+        coordinator: update_coordinator.DataUpdateCoordinator,
+        district: Dict[str, str],
+        info_type: str,
+    ):
+        """Initialize a new sensor."""
+        _LOGGER.debug(f"initialize {info_type} sensor for {district}")
+        super().__init__(coordinator)
+
+        data = coordinator.data[district]
+
+        if data.county:
+            self.name = f"{data.county} {info_type}"
+        else:
+            self.name = f"{data.name} {info_type}"
+        self.unique_id = f"{district}-{info_type}"
+        self.district = district
+        self.info_type = info_type
+        self.updated = datetime.now()
+        self._attr_attribution = f"last updated {self.updated.strftime('%d %b, %Y  %H:%M:%S')} \n{ATTRIBUTION}"
+        self._attr_unit_of_measurement = self._measurement_unit()
+        self._attr_state = self._native_state()
+
+    @property
+    def available(self) -> bool:
+        """Return True if entity is available."""
+        return (
+            self.coordinator.last_update_success
+            and self.district in self.coordinator.data
+        )
+
+    def _native_state(self) -> Optional[str]:
+        """Return current state."""
+        try:
+            return getattr(self.coordinator.data[self.district], self.info_type)
+        except AttributeError:
+            return None
+
+    @property
+    def icon(self):
+        """Return the icon."""
+        return {**SENSORS, **DISTRICT_SENSORS}[self.info_type]
```

Comment:
Remove this and just set `self._attr_icon` in constructor.

Reply:
done